core/modules/search/search.extender.inc

```diff
@@ -534,3 +534,41 @@ class SearchQuery extends SelectQueryExtender {
     return $count;
   }
 }
+
+/**
+ * Extends the core SearchQuery.
+ */
+class viewsSearchQuery extends SearchQuery {
```

Comment:
Not the right person to say if moving this class is OK or not, but code looks OK otherwise.